### ISSUE TITLE
What if there is no answer in the document to the given question?

### ISSUE THREAD

**Issue by maciejbiesek** [HKUST-KnowComp/R-Net#48]
Hi, have you considered training model on dataset with some documents where correct answer to the question does not exist (I mean other dataset, not SQuAD)? How to change the architecture to make it possible? Should we use another loss function?

**Comment by wzhouad** [comment]
Just remove the document in preprocessing, or you can use a zero vector to represent the missing answer and change the loss function to 
```python
loss = tf.reduce_sum(label * -tf.log(tf.nn.softmax(logits) + 1e-6))
```

**Comment by maciejbiesek** [comment]
I am affraid I did not understand. Why should I remove the document in preprocessing? 

Maybe once again I will try to explain: I would like to have model trained with ability to return information that answer to the given question can not be found in this document. For example, we gave the model document about Britney Spears and then asked "Who is Donald Trump?" - obviously there is no chance to answer to that question, so model should return something like "Sorry, I am not able to answer to that".

This model is trained on SQuAD, so there always exists answer as a span of text. But how to train it in case that question and document are not related?

**Comment by wzhouad** [comment]
R-net uses boundary model presented in "Machine Comprehension Using Match-LSTM and Answer Pointer". It will give an answer whether the true answer exists or not. As far as I know no SQuAD model handles this situation. You can try to create a new token "NO ANSWER" and append it at the end of the document in the pointer network layer but I don't know whether it can work.

**Comment by ericchansen** [comment]
Issue #50 may be relevant here. We're working out the best way to return the confidence/probability of a given answer.

If an answer doesn't exist within the context, I imagine that the confidence/probability of the answers that R-NET generates would be low. If the top answer's confidence or probability falls below a certain threshold, you could say, "Sorry, I'm not sure how to answer."

**Comment by javacjh** [comment]
Simple and Effective Multi-Paragraph Reading Comprehension: https://arxiv.org/abs/1710.10723 May help

**Comment by xddai** [comment]
You can read BiDAF code: https://github.com/allenai/bi-att-flow. It concern the situation that there is no answer in the paragraph for some questions.
